### DISCUSSION TITLE
Question on Simulation build: why  OT_REFERENCE_DEVICE=ON and impact on Ping Response

### DISCUSSION THREAD

**Discussion by Alami-Amine** [openthread/openthread#9316] [General]
Hello,

I am wondering why we are activating flag OT_REFERENCE_DEVICE=ON in the OT_simulation build? isn't it supposed to be OFF for a DUT?
the reason i am asking is: I am investigating an issue I have with [MATN_02_MLRFirstUse.py ](https://github.com/openthread/openthread/blob/main/tests/scripts/thread-cert/border_router/MATN/MATN_02_MLRFirstUse.py) and I see that the behavior of this test changes when we have OT_REFERENCE_DEVICE=ON when using Posix platforms, with regards to Ping Replies:

`#if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
    otIcmp6SetEchoMode(gInstance, OT_ICMP6_ECHO_HANDLER_ALL);
#else
    otIcmp6SetEchoMode(gInstance, OT_ICMP6_ECHO_HANDLER_DISABLED);
#endif`

 Basically my issue is that when running the equivalent of [MATN_02_MLRFirstUse.py ] on thread Harness, and using an ot-br-posix as a DUT with Router role, the test fails when  OT_REFERENCE_DEVICE=OFF. and the symptom is : 
I have is exactly the same as that described in this [issue ](https://github.com/openthread/openthread/issues/7175#issuecomment-972613573).
in the table summarized by @simonlingoogle, I am in R2, Certification Column. Basically I can only get the test to pass when i manually add a route for the ping response in my TUN interface (ip -6 route add 910b::/64 dev wpan0)

Any help or clarification on issue described above?

**Comment by abtink** [comment]
The chnage was added in 
- https://github.com/openthread/openthread/pull/7288.

There seems to some related earlier discussion on this:
- https://github.com/openthread/ot-reference-release/pull/20
- https://github.com/openthread/openthread/issues/7175

I don't recall details, @edmont do you happen to remember any points related to this.

**Comment by edmont** [comment]
Hi:

- https://github.com/openthread/openthread/pull/7288 is needed for the reference devices to be able to handle ICMPv6 echo and pass the certification tests as DUT.
- Regular border router devices (non-reference) should enable `BORDER_ROUTING` instead.

@aalami90 is your device failing to pass the test even with `BORDER_ROUTING` enabled?

**Comment by EskoDijk** [comment]
@edmont @abtink  Thanks for clarifying. This will be further discussed in the TSTBED-275 JIRA.

**Comment by EskoDijk** [comment]
Final note, just to record what I think is the outcome on having a "reference device" exhibit different behavior than a "normal" non-reference device:

* the intention of OT / OTBR in general is that having a device into “reference device mode” should never by default change its behavior, compared to a regular non-reference device.
* if that does happen, it’s a bug or oversight of the implementer
* the reference device will expose additional (CLI) commands or options (available for reference devices only) that will allow a test harness or an operator to configure changes in the default behavior, including behaviors that go against the spec.

@edmont @abtink Is this correct?

**Reply by jwhui** [reply]
I believe this is all correct.

**Reply by abtink** [reply]
Yes. Correct.
